src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/SignatureBuilder.cs

```diff
@@ -310,6 +310,12 @@ public void EmitTypeSignature(TypeDesc typeDesc, SignatureContext context)
                 case TypeFlags.ValueType:
                 case TypeFlags.Nullable:
                 case TypeFlags.Enum:
+                    if (typeDesc == PrimitiveTypeProvider.GetPrimitiveType(typeDesc.Context, PrimitiveTypeCode.TypedReference))
```

Comment:
Nit: this is more compact/readable (same compact form can be used for the diff in SignatureContext.cs)

```suggestion
                    if (typeDesc.IsWellKnownType(WellKnownType.TypedReference))
```

Reply:
Thanks for pointing that out, fixed in 2nd commit along with one more place that was using the same check.